Restructure example/trl/stack_llama_2 for generic DPO (#635)
huggingface/optimum-habana: examples/trl/README.md

$ pip install -U -r requirements.txt
```

## DPO pipeline

### Training

> The following example is for the creation of StackLlaMa 2: a Stack exchange llama-v2-7b model
> There were two main steps to the DPO training process:

Reviewer: ```suggestion
The following example is for the creation of StackLlaMa 2: a Stack exchange llama-v2-7b model.
There are two main steps to the DPO training process:
```

        --warmup_steps=100 \
        --weight_decay=0.05 \
        --optim="paged_adamw_32bit" \
        --lora_target_modules "q_proj" "v_proj" \
        --bf16 \
        --remove_unused_columns=False \
        --run_name="sft_llama2" \
        --report_to=none \
        --use_habana \

>         --use_lazy_mode \

Reviewer: ```suggestion
        --use_lazy_mode
```

        --warmup_steps=100 \
        --weight_decay=0.05 \
        --optim="paged_adamw_32bit" \
        --lora_target_modules "q_proj" "v_proj" \
        --bf16 \
        --remove_unused_columns=False \
        --run_name="sft_llama2" \
        --report_to=none \
        --use_habana \

>         --use_lazy_mode \
> 
>     ```

Reviewer: ```suggestion
        --use_lazy_mode
    ```
```

which will also push the model to your HuggingFace hub account.

### Running the model

> We can load the DPO-trained LoRA adaptors which were saved by the DPO training step and run it through example/text-generation folder.

Reviewer: ```suggestion
We can load the DPO-trained LoRA adaptors which were saved by the DPO training step and run it through the [text-generation example](https://github.com/huggingface/optimum-habana/tree/main/examples/text-generation).
```
